scripts/psycorona_modelling.R

```diff
@@ -69,6 +73,77 @@ preds        <- names(df_analyse)[-(which(names(df_analyse) %in% outcome_vars))]
 # Explore outcome variable
 hist(df_analyse$c19perbeh)  # people tend to self-serve
 
+
+########## CARE MODEL COMPARING SETUP ########## 
+
+# make the CV split
+cv_split <- trainControl(method="cv", number=10, verboseIter=TRUE)
+# Train the LASSO 
+lambda <- seq(0.0001,0.01,by = 0.0001)
+# set seed
+set.seed(100)
+caret_lasso <- train(c19perbeh~., data=df_analyse, 
+                     method="glmnet", 
+                     trControl=cv_split, 
+                     family = "gaussian", 
+                     tuneGrid = expand.grid(alpha = 1, lambda = lambda), 
+                     verbose=TRUE)
+
+caret_lasso$bestTune$lambda
+
+# lasso coefficients
+lasso_coef_names <- rownames(coef(caret_lasso$finalModel, caret_lasso$bestTune$lambda))[-1] #coef names, without intercept
+lasso_coef_values <- abs(as.vector(coef(caret_lasso$finalModel, caret_lasso$bestTune$lambda))[-1]) #ceof values without intercept
+lasso_coefs_df <- data.frame(coef_names = lasso_coef_names[order(abs(lasso_coef_values), decreasing=TRUE)],
+                             coef_values = lasso_coef_values[order(abs(lasso_coef_values), decreasing=TRUE)])
+
+lasso_coefs_df_excl <- lasso_coefs_df[which(lasso_coefs_df[,2] == 0), 1]    # these features have been excluded
+lasso_coefs_select <- lasso_coefs_df[-(which(lasso_coefs_df[,2] == 0)), ] # these features have been selected (= meaningful)
+
+# Coef plot
+ggplot(lasso_coefs_df,
+       aes(x = 0, xend = coef_values, y = reorder(coef_names, coef_values), yend = coef_names)) + 
+  geom_segment() +
+  geom_point(aes(x = coef_values, y = coef_names)) +
+  theme_bw()
+
+
+# Training RANDOM FOREST
+mtry <- round(sqrt(ncol(df_analyse)))
+# set seed
+set.seed(100)
+caret_rf <- train(c19perbeh~., data=df_analyse, 
+                  method="rf",  
+                  ntree = 30,
```

Comment:
500 is usually the minimum, but I imagine this takes a long time on your PC?

Reply:
Yeah, even this already took some time to compute :(. 500 was really unfeasible for my laptop.